API_DESIGN.md

```diff
@@ -39,6 +39,12 @@ Please check out their style guides and concepts for more information:
 Additionally, there are some conventions that are specific to the ZITADEL API.
 These conventions are described in the following sections.
 
+### File Structure
+
+In addition to the files structure proposed by Protobuf and connectRPC, the ZITADEL API uses the following order for `Everything else` section of files:
+- `Service(s)`
+- `Messages` / `Enums`
```

Comment:
would it make sense to add some info about ordering of requests ? e.g. reading > creation > update > remove

and mention that messages should have the order of the requests